File: DataStructures/Graphs/creation/graphs.c
#include <stdio.h>
#include <stdlib.h>

#define V 5  // Number of vertices

struct Node {
    int data;
    struct Node* next;
};

// Array of pointers to nodes (Adjacency List)
struct Node* graph[V];

void addEdge(int src, int dest) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = dest;
    newNode->next = graph[src];
    graph[src] = newNode;
}

int main() {
    // Initialize all adjacency lists as empty
    for (int i = 0; i < V; i++) graph[i] = NULL;

    addEdge(0, 1);
    addEdge(0, 4);
    addEdge(1, 2);
    addEdge(1, 3);
    addEdge(1, 4);
    addEdge(2, 3);
    addEdge(3, 4);

    // No traversal or printing
    return 0;
}

File: DataStructures/Graphs/creation/graphs.py
from collections import defaultdict

graph = defaultdict(list)

def add_edge(graph, src, dest):
    graph[src].append(dest)

# Adding edges to the graph
add_edge(graph, 0, 1)
add_edge(graph, 0, 4)
add_edge(graph, 1, 2)
add_edge(graph, 1, 3)
add_edge(graph, 1, 4)
add_edge(graph, 2, 3)
add_edge(graph, 3, 4)

